### ISSUE TITLE
Dapper.Contrib does not respect SqlMapper.TypeMaps

### ISSUE THREAD

**Issue by SwissMaWi** [StackExchange/Dapper#1177]
The Dapper.Contrib does not use the ITypeMap instances set for mapping database table column names to POCO property names. Generated SQL statements therefore are wrong when using a POCO with property names not identical to database columns.

**Comment by SwissMaWi** [comment]
Proposal: Modify Dapper.Contrib.SqlMapperExtensions to respect any ITypeMap which have been set by using the SetTypeMap method.

**Comment by SwissMaWi** [comment]
I am trying to implement this. However: I do not seem to be able to run the Tests. There is a database server required. Can I get any support about how to run the tests in the Dapper solution?

**Comment by SwissMaWi** [comment]
I am trying to implement the reverse mapping and make Dapper.Contrib use it to create correct SQL statements when an ITypeMap is present for a type. Do you have any hints for me @NickCraver I am fairly new to this. Would you be willing to review? Regards, Martin

**Comment by Brondahl** [comment]
What happened with this. This is causing my project some pretty painful problems?

It looks like #722 is about a change in DapperV2, which I doubt we'd be moving to, even if it were released tomorrow (stodgy corporate environment :( ).

Is there scope for fixing this issue in Dapper.Contrib directly?

(We could probably live with pulling in the DapperContrib source code and monkey patching it ourselves, if that's the only option - is this a reasonably accessible change to make, or does it require large restructuring?)

**Comment by Brondahl** [comment]
@SwissMaWi Did you ever resolve this? Did you identify the correct fix to apply?

**Comment by Brondahl** [comment]
@NickCraver Is my understanding correct - that #722 is going to be a new major version of Dapper?

**Comment by Jetski5822** [comment]
Darn - just hit this too.. :(

**Comment by eliavra** [comment]
is there plans to fix this issue?
if not, is there a workaround to using the mapper ?

**Comment by rclarke2050** [comment]
could we have a beta version (pre-release) at least to go with, so we can at least move forward a little with our projects.  The  MyFieldName {get;set;}  -->  <table>.myfieldname   is causing a big headache for me now, because even though for SqlConnection the class names are fine, for postgres its not.  The TypeMap idea is good Nick, but i can see other people have posted their solutions, https://github.com/StackExchange/Dapper/pull/1396, 
all we need is beta around this issue for the time being until DapperV2 comes out.  is that do'able?

**Comment by rclarke2050** [comment]
I found a way around this field name problem for the time being.

in my Repository Constructor i used:

DapperExtensions.DapperExtensions.SqlDialect = new DapperExtensions.Sql.PostgreSqlDialect();

and in the Insert Method:   db.Insert(item); 
and in the Updaet Method: db.Update(item);

i did not use the SqlMapper approach, nor did I use the SqlMapperExtensions.Insert()

Ive tried on .Net Core 2.1 Lambda, and AWS Aurora 10.7.

Dapper v2.0.30
Dapper.Contrib 2.0.30
DapperExtensions.DotnetCore 1.0.1

I hope this helps others for now....in the interim.

**Comment by vincentparrett** [comment]
I've come to the conclusion that Dapper.Contrib is not currently usable for postgresql - not using the typemap is a major deal since column names are case sensitive.

**Comment by rclarke2050** [comment]
> I've come to the conclusion that Dapper.Contrib is not currently usable for postgresql - not using the typemap is a major deal since column names are case sensitive.

This is not true, I have managed to get around the issue, aslong as the column name appending on the PostgresAdapter is sorted (which i believe is now), then you wont have the problem with direct class property to db column mapping out of the box.  

My aws Aurora Postgres db is defaulted to case sensitive, and its working flawlessly.

my 50c
